source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2932,8 +2727,9 @@ void COpenGLDriver::clearZBuffer(const float &depth)
         return;
 
 
-    glDepthMask(GL_TRUE);
-    LastMaterial.ZWriteEnable=true;
+    found->nextState.rasterParams.depthWriteEnable = 1;
+    //TODO or maybe i can just assume it should be enabled at this point?
```

Comment:
correct buffer clears on opengl wont take place if write masks are not enabled

Reply:
also necessar for color clears

Reply:
@Crisspl all the other `COpenGLDriver::clear*` need similar state setting

Reply:
you also need to disable rasterizer discard state

Reply:
what has to be enabled/disabled for stencil clears to work? and for color clears as well? something beside write masks?

Reply:
disable:
- rasterizer discard
- scissor test
enable:
- color write mask (color clear)
- depth write mask (depth clear)
- stencil write mask both front and back (stencil clear)

Reply:
resolved

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r328449492